cinema/views.py

```diff
@@ -1 +1,82 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import (
+    Genre,
+    Actor,
+    Movie,
+    MovieSession,
+    Order,
+    Ticket,
+    CinemaHall
+)
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    MovieSessionSerializer,
+    OrderSerializer,
+    TicketSerializer,
+    CinemaHallSerializer,
+    MovieSerializer,
+    MovieSessionListSerializer,
+    MovieListSerializer,
+    MovieSessionRetrieveSerializer,
+    MovieRetrieveSerializer
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+    def get_queryset(self):
+        if self.action in ("list", "retrieve"):
+            return self.queryset.select_related()
+        return self.queryset
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+
+
+class TicketViewSet(viewsets.ModelViewSet):
+    queryset = Ticket.objects.all().select_related()
```

Comment:
The `TicketViewSet` uses `.select_related()` on the queryset. Ensure that this call is necessary and not redundant, as it may affect performance. If it's redundant, consider removing it.